### ISSUE TITLE
Build-time tests aren't visible

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#32603]
### Steps to reproduce

Choose any package for which the build-time tests fail. An easy way to reproduce this is to choose any Python package with minimal dependencies and override `import_modules` to be a list of fake modules:
```python
import_modules = ["foo", "bar"]
```
Then, try to run the build-time tests:
```console
$ spack install -v --test=root foo
```

### Error message

You'll see the full build log, but no build-time tests. All you'll see if that _something_ failed, but there's no way to tell _what_ failed. This is not specific to Python's import tests, it affects all packages. For example, here is the output from an Autotools package:
```
make[4]: Nothing to be done for `all'.
make[4]: Nothing to be done for `all-am'.
make[2]: Nothing to be done for `all-am'.
==> Testing package gnutls-3.6.15-xbt76kt
==> [2022-09-11-10:59:24.365028] Running build-time tests
==> [2022-09-11-10:59:24.365144] ('RUN-TESTS: build-time tests [check]',)
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'check'
```
I don't know _why_ `make check` failed, and I don't see any way to find out this information since it isn't visible in the terminal or in the build log. This is not the way things used to be. I would have to do a git bisect to figure out exactly when it changed, but I'm guessing it is related to recent post-install test refactors.

### Information on your system

* **Spack:** 0.19.0.dev0 (16655f1be20ef001851cb5998b0f3e9b3df33147)
* **Python:** 3.9.13
* **Platform:** darwin-monterey-m1
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by adamjstewart** [comment]
Duplicate of #31920

**Comment by tldahlgren** [comment]
I tried this and am seeing details in the `install-time-test-log.txt` file in the build stage directory for the package.  Are you not seeing a file with that name?

**Comment by adamjstewart** [comment]
I've never heard of this file before. I'm sure it's there, and that's good to know. But this issue is suggesting that `spack install --test --verbose` should display that output just like it does with the build output. Otherwise I have no idea that the tests are actually running vs. being skipped. That's how things used to be before post-install tests were added.

**Comment by tldahlgren** [comment]
> I've never heard of this file before. I'm sure it's there, and that's good to know. But this issue is suggesting that `spack install --test --verbose` should display that output just like it does with the build output. Otherwise I have no idea that the tests are actually running vs. being skipped. That's how things used to be before post-install tests were added.

Good point.  I'll look into outputting the path to that file on success and failure.

**Comment by adamjstewart** [comment]
We should def output the path on failure (like we do for build). But we should output the contents when verbose as well (like we do for build).
